proxy/src/main/java/net/md_5/bungee/UserConnection.java

```diff
@@ -252,6 +252,11 @@ public void connect(ServerInfo info, final Callback<Boolean> callback, final boo
             {
                 callback.done( false, null );
             }
+            if ( server == null )
+            {
+                String s = bungee.getTranslation( "connect_kick", info.getName() );
+                disconnect( s.substring( 0, s.length() - 1 ) ); // Remove the colon at end of string
```

Comment:
"if ( server == null )" Variable 'info' is the server they are trying to connect to